build.gradle.kts

```diff
@@ -17,41 +15,47 @@ repositories {
     mavenCentral()
 }
 
-dependencies {
-    api("org.jetbrains.kotlinx:kotlinx-serialization-json:1.4.1")
-    testImplementation(kotlin("stdlib"))
-    testImplementation(kotlin("test"))
-    testImplementation("org.junit.jupiter:junit-jupiter-api:5.9.0")
-    testRuntimeOnly("org.junit.jupiter:junit-jupiter-engine:5.9.0")
-}
-
-java {
-    sourceCompatibility = JavaVersion.VERSION_1_8
-    targetCompatibility = JavaVersion.VERSION_1_8
-    withJavadocJar()
-    withSourcesJar()
-}
-
-tasks {
-    withType<KotlinCompile>().configureEach {
-        this.kotlinOptions {
-            jvmTarget = "1.8"
-            freeCompilerArgs = listOf("-opt-in=kotlinx.serialization.ExperimentalSerializationApi")
+kotlin {
+    jvm {
+        compilations.all {
+            kotlinOptions {
+                jvmTarget = JavaVersion.VERSION_1_8.toString()
+            }
         }
     }
+    linuxX64()
+    mingwX64()
+    macosX64()
```

Comment:
On Windows 10 x64 I get:
```
Some Kotlin/Native targets cannot be built on this mingw_x64 machine and are disabled:
    * In project ':':
        * target 'macosX64' (can be built with one of the hosts: macos_x64, macos_arm64)
```
will the CI be able to build all targets?

Reply:
Yep, this applies to linux CI too. This means the full project is only buildable on Mac?

Reply:
yes, macos builds all targets. 
Can I change workflows from `runs-on: ubuntu-latest` to    `runs-on: macos-latest`?

Reply:
> On Windows 10 x64 I get:

btw, I got failed tests on windows in `serialize kotlin data class to json`. The reason is in pretty print in JSON and end of line which is platform dependent. 
Test can be fixed by splitting String to lines().

Is it okay to add it?

Reply:
I would raise a separate PR for that, can be done even from GitHub UI.

Reply:
done: https://github.com/detekt/sarif4k/pull/7/files

Reply:
Since the CI is set up correctly, shall we add the ignore for this message so it doesn't bother everyone every time who builds this?

> To disable this message, add 'kotlin.native.ignoreDisabledTargets=true' to the Gradle properties.

Reply:
Added in https://github.com/detekt/sarif4k/pull/6/commits/a36a8d34873444c428c6f15f64e49f97699a5156